torchvision/transforms/functional.py

```diff
@@ -105,9 +105,17 @@ def to_pil_image(pic, mode=None):
     Returns:
         PIL Image: Image converted to PIL Image.
     """
-    if not(_is_numpy_image(pic) or _is_tensor_image(pic)):
+    if not(torch.is_tensor(pic) or isinstance(pic, np.ndarray)):
         raise TypeError('pic should be Tensor or ndarray. Got {}.'.format(type(pic)))
 
+    elif torch.is_tensor(pic):
+        if pic.ndimension() != 3:
+            raise ValueError('pic should be 3 dimensional. Got {} dimensions.'.format(pic.ndimension()))
```

Comment:
Can you remind me again why the code works for 2d np arrays, but not 2d tensors?

Reply:
The `is_numpy_image` function checks for both 2D and 3D arrays. I assumed this was for some other functionality and left that in to avoid breaking something else. I can add code to work for 2D tensors as well if you'd like.

Reply:
I see. But there was a bug before for 2d arrays, right?

Reply:
I don't think so. The bug has been for 2D tensors - we get a TypeError exception.